pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerService.java

```diff
@@ -138,6 +148,14 @@ public void start(URI dlogUri) throws InterruptedException {
             // indicate function worker service is done intializing
             this.isInitialized = true;
 
+            this.connectorsManager = new ConnectorsManager(workerConfig);
+            
+            int metricsSamplingPeriodSec = this.workerConfig.getMetricsSamplingPeriodSec();
+            if (metricsSamplingPeriodSec > 0) {
+                this.statsUpdater.scheduleAtFixedRate(() -> this.functionRuntimeManager.updateRates(),
```

Comment:
Why does this need to be done here? why can't each instance periodically reset their metrics themselves?